consensus/types/src/light_client_header.rs

```diff
@@ -15,12 +21,160 @@ use test_random_derive::TestRandom;
     TestRandom,
     arbitrary::Arbitrary,
 )]
-pub struct LightClientHeader {
+#[serde(bound = "E: EthSpec")]
+#[arbitrary(bound = "E: EthSpec")]
+pub struct LightClientHeader<E: EthSpec> {
     pub beacon: BeaconBlockHeader,
+    #[test_random(default)]
+    #[ssz(skip_serializing, skip_deserializing)]
+    pub execution: Option<ExecutionPayloadHeader<E>>,
```

Comment:
im skipping serializing here, because Option is causing errors. Theres probably a way to fix this, I just haven't figured it out yet

Reply:
Can you try to use superstruct here? Since fields are all fork dependant it fits perfectly to do the job. At least give it a try on another branch and report here which blockers are you encountering.

Reply:
yep np, i'll go ahead and use superstruct